mllam_verification/plot.py

```diff
@@ -0,0 +1,46 @@
+import matplotlib.pyplot as plt
+import xarray as xr
+import mllam_verification as mlverif
+ 
+ 
+def plot_single_metric_timeseries(
+    ds_reference: xr.Dataset,
+    ds_prediction: xr.Dataset,
+    variable: str,
+    metric: str,
+    axes: plt.Axes = None,
+    include_persistence=True,
+    xarray_plot_kwargs: dict = {},
+):
+    """Plot a single-metric-timeseries diagram for a given variable and metric.
+ 
+    The metric is calculated from ds_reference and ds_prediction.
+     
+    Parameters
+    ----------
+    ds_reference : xr.Dataset
+        Reference dataset.
+    ds_prediction : xr.Dataset
+        Prediction dataset.
+    variable : str
+        Variable to calculate metric of.
+    metric : str
+        Metric to calculate.
+    axes : plt.Axes, optional
+        Axes to plot on, by default None
+    xarray_plot_kwargs : dict, optional
+        Additional arguments to pass to xarray's plot function, by default {}
+    """
+ 
+    ds_metric = mlverif.calculate_{metric}(ds_reference, ds_prediction, variable, include_persistence=include_persistence)
+ 
+    if axes is None:
+        axes = mlverif.operations.plot_utils.get_axes(plot_type="timeseries")
+ 
+    ds_metric[metric].plot.line(ax=axes, **xarray_plot_kwargs)
+ 
+    if include_persistence:
+        ds_metric["persistence"].plot.line(ax=axes, **xarray_plot_kwargs)
```

Comment:
Maybe the metric calculation could add a new coordinate to the dataset called say `data_source` which could include names like `[persistence, DINI_forecast, ...]` (`persistence` would only be there if `include_persistence==True`, the other names could be set from dict keys if we pass in multiple prediction datasets), then plotting could be replaced with:

```suggestion
    ds_metric[metric].plot.line(ax=axes, hue="data_source", **xarray_plot_kwargs)
```

Reply:
I don't think we should need to select the `metric` variable here, either we compute the same metric for all variables and select a specific variable here, or only calculate the metric for a specific variable, in which case the return from the metric function could just be a data-array

Reply:
Regarding my comment above, maybe we should rather than using a dict require that the user does concatenation along a `data_source` dimension if the `ds_prediction` argument is to contain multiple sources

Reply:
> I don't think we should need to select the `metric` variable here, either we compute the same metric for all variables and select a specific variable here, or only calculate the metric for a specific variable, in which case the return from the metric function could just be a data-array

Agree, the `[metric]` part is not needed. I think we should stick with only calculating one metric for one variable and let the metric function return a data-array then.

Reply:
> Regarding my comment above, maybe we should rather than using a dict require that the user does concatenation along a `data_source` dimension if the `ds_prediction` argument is to contain multiple sources

We could do that yet. Not sure which one is the better design.